dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/patch/AbstractResourcePatch.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository.patch;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.UUID;
+
+import org.dspace.app.rest.exception.PatchBadRequestException;
+import org.dspace.app.rest.exception.UnprocessableEntityException;
+import org.dspace.app.rest.model.patch.Operation;
+import org.dspace.app.rest.model.patch.Patch;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.core.Context;
+
+/**
+ * The base class for resource PATCH requests.
+ *
+ * @author Michael Spalti
+ */
+public abstract class AbstractResourcePatch {
```

Comment:
This class should use generics to deal with the RestModel instead than apiCategory, model and ID
i.e. `AbstractResourcePatch<R extends RestModel>` and use `R restModel` instead than the three arguments in the subsequent methods